src/GitHub/Lens.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE PolyKinds     #-}
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+{- |
+Copyright: (c) 2021 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Utilities to provide general @lens@-like API for creating GraphQL
+queries.
+-}
+
+module GitHub.Lens
+    ( HasOwnerName (..)
+    , HasLimit (..)
+    , HasStates (..)
```

Comment:
We maybe should consider two distinct naming schemes: one for lenses, another for data like `Title`.
I am not sure, what should go where, but maybe it is easier to mark lens-typeclasses with `L` siffix? Or maybe you have any other ideas? 🤔